cpp/src/gandiva/projector.cc

```diff
@@ -186,7 +192,10 @@ Status Projector::Make(SchemaPtr schema, const ExpressionVector& exprs,
   // Instantiate the projector with the completely built llvm generator
   *projector = std::shared_ptr<Projector>(
       new Projector(std::move(llvm_gen), schema, output_fields, configuration));
-  cache.PutModule(cache_key, *projector);
+  ValueCacheObject<std::shared_ptr<Projector>> value_cache =
+      *std::make_unique<ValueCacheObject<std::shared_ptr<Projector>>>(*projector,
```

Comment:
This is not correct. You are creating a temporary shared pointer and then immediately dereferencing it, which will lead to destructor getting called after the statement.
And why the need to unnecessarily create a temporary shared pointer? just do normal stack initialization.